### ISSUE TITLE
Call volume too low

### ISSUE THREAD

**Issue by ComeGitSome** [WhisperSystems/Signal-Android#4244]
TextSecure 3.0.3 on Samsung Galaxy S4 with Android 5.0.1

The call feature is currently not usable due to very low volume making it hard to hear the other person.

**Comment by 2-4601** [comment]
Does this happen every time on every call?
So far I have experienced this only once when a Signal iOS user called me (3.0.3). I was able to fix it during the call but not sure what exactly solved it. I played around with the system volume settings and tapped the speaker mode on and off in Signal. I have not been able to reproduce this.

**Comment by janvlug** [comment]
See also: https://github.com/WhisperSystems/RedPhone/issues/245

**Comment by ComeGitSome** [comment]
Happens every time. Always happened with Redphone as well.

**Comment by jeremymasters** [comment]
I think overall a 30% to 35% boost in volume would be about right.

**Comment by johanw666** [comment]
I see this comment in RedPhone.java in the RedPhoneproject, but can't find it in the Signal version, maybe it has something to do with this issue?
```java
 @Override
public boolean onKeyDown(int keyCode, KeyEvent event)
{
  boolean result = super.onKeyDown(keyCode, event);
  //limit the maximum volume to 0.9 [echo prevention]
  if (keyCode == KeyEvent.KEYCODE_VOLUME_UP) {
    AudioManager audioManager = (AudioManager)
...
```

**Comment by 2-4601** [comment]
Happened again. A Signal iOS user (2.1.3, iPhone 4S) called me (Signal 3.0.3, Nexus 4, stock 5.1.1) and the voice was barely audible. After tapping the speaker icon on and off the voice level normalized.

**Comment by Dyras** [comment]
I have a Galaxy Nexus and I have this exact problem while talking to an Xperia S. I can barely hear shit unless the room is 100% quiet.

**Comment by b0m0x** [comment]
Same Problem on a Sony device. I had it before with RedPhone and it's still there after updating to Signal.

**Comment by 2-4601** [comment]
I can consistently "fix" this by putting the call on the speaker and then back to the handset.

**Comment by moxie0** [comment]
I just ordered a S4, will take a look once I get it and can reproduce.

**Comment by Dyras** [comment]
@moxie0 Which phone are you using right now, that doesn't have this problem? It makes me curious, considering how almost everyone seems to have this problem.

**Comment by johanw666** [comment]
Maybe it would be more objective to compare the call loudness to the loudness of a normal voice call on the same device. Some people might have better hearing than others, "too low" is subjective.

**Comment by Dyras** [comment]
"Ordinary" calls I can hear just fine.
Signal calls I can hear if the room is completely silent and no one is talking.

**Comment by digititus** [comment]
This happens on every Signal call I have from and to Android. Never happened on Redphone. Tested with 2  moto G's, Xerperia Z3 and iPhone 4s. In public spaces, the calls are basically inaudible.

**Comment by x2k13** [comment]
Same for me here. 
Honor 6 (Android 5.1.1) TS/Signal 3.1.1 and iPhone 5 (iOS 9.1) Signal 2.1.3.
No matter who is calling who, everytime the call volume is set too low (on Android) and I have to manually tick the speaker button on and off to get the volume up to normal again.

**Comment by ryannathans** [comment]
I have this problem too.

Samsung Note 4 <-> Note 3 = Note 3 really quiet, Note 4 normal volume
Samsung Note 4 <-> S5 = Note 4 really quiet, S5 normal volume

No amount of volume changing, redialing or changing to speakerphone and back will resolve the issue.

Had this issue in the past with AGC

**Comment by ComeGitSome** [comment]
Signal 3.1.1 on Samsung Galaxy S4 with Android 5.0.1

Sound volume has been normal the last few calls. But last call the other person, using an iPhone, had the problem with low volume.

**Comment by vinilox** [comment]
Signal 3.1.1 on Nexus 4 with Android 5.1.1 and Moto G (2nd Gen) with Android 5.0.x.

Applies to both phones : Never called or been called by somebody with an iPhone but I have this problem. I can't use encrypted calls because of this. It's a very low volume.

**Comment by jeremymasters** [comment]
@vinilox I think you can hit the speaker or mic button and hit it again to get normal volume.  Worth a shot anyway.

**Comment by vinilox** [comment]
I will try thank you ! :)

Le mardi 03 novembre 2015 à 08:02 -0800, jeremymasters a écrit :

**Comment by jocelynthode** [comment]
I never had this problem on Redphone and have it on Signal 3.1.1 as weel

Samsung Galaxy i9300

**Comment by darryl-grennan** [comment]
I've been testing TextSecure and RedPhone for a while now.
I see you've merged the two.

My phone: Samsung Galaxy S6 / Model: SM-G920W8 running Android 5.1.1

I get extremely *low audio* during voice calls through the earpiece speaker
placed with Signal as I did with RedPhone. *Even when the volume is
cranked all the way up.* It also sounds tinny (lower sound frequencies
aren't reproduced properly).

Can this please get fixed? I've tried this on other Android phones and the
same problem persists. **This is a deal breaker for a lot of my friends who
have uninstalled it for this very reason.**

**Comment by SlappyHours** [comment]
Same problem here. Also the sound quality could be better.

**Comment by ogruetzmann** [comment]
Same issue on Oneplus One.

**Comment by ogruetzmann** [comment]
Forgot to mention, just checked on Signal 3.1.1

**Comment by moxie0** [comment]
Should be fixed in 3.3.0.  If anyone wants to test and confirm, would appreciate it.

**Comment by jeremymasters** [comment]
@moxie0 I could test it.  Let me know.

**Comment by darryl-grennan** [comment]
I'm in the beta but I suspect we'd need to build this to test.  Are there APK test builds created somewhere?

**Comment by 2-4601** [comment]
Well, I tried 0e291a9509bb364b7595b88dae452df7c714ed7a but forgot that I didn't have this issue with my test device in the first place.
However I noticed other issues:
- The call quality is noticeably worse after this commit (I tested several times switching between different builds and it is consistent)
- When I adjust volume during call there is a bluetooth icon in the overlay volume screen instead of the normal phone icon
Perhaps it would be more clear to make a new issue about these.
Huawei U8800, Android 4.0.4

**Comment by Dyras** [comment]
@moxie0 Push a new beta with the update and you will find out very, very quickly if it helped or not.

**Comment by 2-4601** [comment]
Testing between Huawei U8800 and Nexus 4 with the final 3.3.0 release.
The Huawei has the bad audio quality as I have mentioned before. However Nexus 4 has good quality voice but the volume level is too low. And the speaker on/off trick does not help any more. Well, at least it's not as inaudible as most of the cases have reported here. But still it's too low.

**Comment by darryl-grennan** [comment]
I can't even make calls now.
I get a white and blue blank screen.

The other person I'm attempting to call gets random words placed on their screen.  (Ammo Infancy).

Here's the debug URL: https://gist.github.com/09bfc1cb8efe9aafdd64

![img_20151105_193230_617](https://cloud.githubusercontent.com/assets/15638229/10986231/3c242d70-83f6-11e5-9dfb-cafa86c3d65d.jpg)
![img_20151105_193753_087](https://cloud.githubusercontent.com/assets/15638229/10986234/4289c77e-83f6-11e5-9621-3c2b373e7fc9.jpg)

**Comment by 2-4601** [comment]
@darryl-grennan This sounds like a bug that is not related to low call volume. Would you mind opening a new issue for this?

About the two random words, it's a security feature, see the faq: http://support.whispersystems.org/hc/en-us/articles/213608537-What-are-the-two-words-on-my-call-screen-

**Comment by moxie0** [comment]
If everyone who gets the 3.3.1 update that previously had low volume issues could post their 3.3.1 experience here that'd be helpful.

**Comment by digititus** [comment]
@moxie0 Tested 3 android devices. All call volumes were normal with 3.3.1

**Comment by 2-4601** [comment]
Tested calling to Signal Android 3.1.1 and to Signal iOS 2.1.3. Volume levels were fine in every call.

**Comment by darryl-grennan** [comment]
I've tested this version and it appears that volume issues are now fixed.

**Comment by Dyras** [comment]
It's working just fine for me now. No complaints. Well, aside from the front camera not working. Unrelated issue though.

**Comment by danielbald** [comment]
Android 6 Nexus 5 Signal 3.1.1 call volume still too low, switching to speaker was ok. When called Android 5.1.1 Note 5 Edge Signal 3.1.1, volume too low for them both in call and speaker.

**Comment by jocelynthode** [comment]
I had the problem prior to 3.3.1 and I still have it with the 3.3.2 version. As said pressing the speaker volume on/off still fixes the problem.

**Comment by jocelynthode** [comment]
I could also recreate an issue since I don't have the low quality issues that people have but just the low volume.

**Comment by peterpanondrugs** [comment]
My phone (3.4.2) calling other (3.3.2). 
My volume level on Signal is still very low, quite impossible to use outside/city!
I'm using a Samsung S3 LTE / with 5.1.2. (resurrection remix)

Normal calls are way more louder!

**Comment by janvlug** [comment]
In the older Redphone/Signal versions, the call volume was for me too low as well.
Since a few versions the Signal call volume is now OK, but after receiving a Signal call, the volume of a regular phone call is way too high now, like the call is on speaker. Currently I'm on Signal version 3.4.2.

**Comment by Dyras** [comment]
I have the same problem as janvlug. My mom called me and I answered. We talked for like a minute and when I hung up my heart almost stopped when another contact wrote a message to me, because I was wearing a headset and the audio was at the veeeeeeeeeery highest. I think there's a separate issue for that though.

**Comment by codethief** [comment]
The call volume is also too low for me. The person on the other end and I were running v3.4.2.

**Comment by Dyras** [comment]
Yeah I have to say, it's still quite low compared to "ordinary" phone calls.

**Comment by gitdexby** [comment]
Also experiencing the low call volume. It seemed to get louder after I went to speaker mode and then back to regular. The volume really is very low...usable but only if there's no other sound around.

Phone: Sprint LG-LS970
Android: 4.1.2
Signal: 3.5.2

Hope you guys fix this soon. Great product.

**Comment by d-3-n** [comment]
Why is this issue closed? Signal 3.8.1 and the issue is still present. I wonder why just few people complained... Everyone I know has this issue. Different phones (Moto G 1/2 Gen, Galaxy Nexus, Galaxy S4 and 5.) and different ROMs like stock Touchwiz on S5, stock Moto G2 Gen ROM, Cynogenmod 12.1 on S5, S4. Moto G, Cynogenmod 11 on Galaxy Nexus.
Issue happens in all combinations. I thought first it could be related to Cynogenmod (AOSP), but it happens also between stock, not rooted phones, like when calling S5 with original Samsung ROM from Moto G 2nd Gen (also with stock Motorola system). 

As other people mentioned, fiddling with sound Volume helps. Actually I think (In my case.) not fiddling but simply raising the volume to the max. makes the conversation possible (Thought the sound is still quite low.).

**Comment by ryannathans** [comment]
Issue still present for me

Samsung Note 4 android 5.1.1

**Comment by cascheberg** [comment]
Please check if the issue is still present in 3.9.0 if you can. It is available on the Google Play Store as a beta release.

**Comment by ryannathans** [comment]
@cascheberg I can not find it on the play store. Can you please provide a link?

**Comment by d-3-n** [comment]
Christian,

thanks, I registered as a tester and installed 3.9. Will inform you here
about the result later today.

**Comment by d-3-n** [comment]
Register here and update after waiting for few minutes:

https://play.google.com/apps/testing/org.thoughtcrime.securesms/join

**Comment by 2-4601** [comment]
This was closed because there was a commit that fixed this. But as it turned out it worked only for some devices so another issue was opened (#4733) for those that still had low volume in calls.
Yes, it's confusing to have two issues for this.

**Comment by d-3-n** [comment]
it looks like beta fixed the issue for me.

**Comment by agrajaghh** [comment]
ok, lets close this one then :wink: 

everybody: Please open a new issue if you still encouter this in 3.9.0
